keyboards/neopad/rev1/rev1.c

```diff
@@ -0,0 +1,59 @@
+/* Copyright 2021 rookiebwoy
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev1.h"
+
+void keyboard_pre_init_kb(void) {
+    // Set LED IO as outputs
+    setPinOutput(LED_00);
+    setPinOutput(LED_01);
+    keyboard_pre_init_user();
+}
+
+void shutdown_user() {
+    // Shutdown LEDs
+    writePinLow(LED_00);
+    writePinLow(LED_01);
+}
+
+layer_state_t layer_state_set_kb(layer_state_t state) {
+    // Layer LEDs act as binary indication of current layer
+    uint8_t layer = get_highest_layer(state);
+    writePin(LED_00, layer & 0b1);
+    writePin(LED_01, (layer >> 1) & 0b1);
+    return layer_state_set_user(state);
+}
```

Comment:
This might be better off as: 
```suggestion
layer_state_t layer_state_set_kb(layer_state_t state) {
    state = layer_state_set_user(state);
    
    // Layer LEDs act as binary indication of current layer
    uint8_t layer = get_highest_layer(state);
    writePin(LED_00, layer & 0b1);
    writePin(LED_01, (layer >> 1) & 0b1);
    return state;
}
```

Reply:
Thank you for your suggestion. May I ask you if this is only for cleaning the code or if my first try can produce some unexpected behaviors that I didn't detect?

Reply:
It's so that the user function is run first, and gets any changes from that, first.